scholia/app/views.py

```diff
@@ -130,6 +131,117 @@ def redirect_q(q):
     return redirect(url_for(method, q=q), code=302)
 
 
+@main.route("/" + q_pattern + "/bioschemas")
+def bioschemas_for(q):
+    """Detect a Bioschemas API request and return Bioschemas for this item.
+
+    Parameters
+    ----------
+    q : str
+        Wikidata item identifier
+    """
+    entity_class = q_to_class(q)
+
+    # get the aspect specific bits
+    if entity_class == "author":
+        data = {
+            "@type": "Person",
+            "http://purl.org/dc/terms/conformsTo": {
+                "@type": "CreativeWork",
+                "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19/",
+            },
+            "description": "A person",
+        }
+    elif entity_class == "substance":
+        data = {
+            "@type": "ChemicalSubstance",
+            "http://purl.org/dc/terms/conformsTo": {
+                "@type": "CreativeWork",
+                "@id": "https://bioschemas.org/profiles/ChemicalSubstance/0.4-RELEASE/",
+            },
+        }
+    elif entity_class == "taxon":
+        data = {
+            "@type": "Taxon",
+            "http://purl.org/dc/terms/conformsTo": {
+                "@type": "CreativeWork",
+                "@id": "https://bioschemas.org/profiles/Taxon/0.6-RELEASE/",
+            },
+        }
+        taxon_name = property_for_q(q, "P225")
+        if taxon_name:
+            data['name'] = taxon_name
+        rank = property_for_q(q, "P105")
+        if rank:
+            data['taxonRank']: rank
+        taxon_parent = property_for_q(q, "P171")
+        if taxon_parent:
+            data['taxonParent'] = taxon_parent
+    elif entity_class == "chemical":
+        data = {
+            "@type": "MolecularEntity",
+            "http://purl.org/dc/terms/conformsTo": {
+                "@type": "CreativeWork",
+                "@id": "https://bioschemas.org/profiles/MolecularEntity/0.5-RELEASE/",
+            },
+        }
+        inchi_key = property_for_q(q, "P235")
+        if inchi_key:
+            data['inChIKey'] = inchi_key
+        if inchi:
```

Comment:
it seems I may have lost the query for the inchi string

Reply:
no worries. I want to add `properties_for_q(q, ["P235", "Pxx"])` first anyway (return a map)

Reply:
i was just thinking about how that would be nice!

Reply:
the `inchi` is now fetched, but still in the old way.

Reply:
The method is now implemented: https://github.com/WDscholia/scholia/pull/1478/commits/3ef5ba659245a1dfbcc202f98f3cc9fbf6aac885